digAddressExtractor/address_extractor.py

```diff
@@ -0,0 +1,204 @@
+
+def cleanAddress(text_string,level):
+    if level>0:
+        #Slicing from 'location' word in found adddress
+        pos=text_string.find('location')
+        if pos >-1:
+            text_string=text_string[pos+len('location'):]
+
+    if level>1:
+        #Slicing from phone number reference word in found adddress to end
+        m=phonePattern.search(text_string)
+        if m!=None:
+            pos=m.span()[1]
+            text_string=text_string[pos:]
+
+    if level>2:
+        #Cleaning if maps URL present
+        if text_string.find('maps.google.com') >-1 or text_string.find('=')>-1:
+            pos=text_string.rfind('=')
+            if pos >-1:
+                text_string=text_string[pos+1:].replace('+',' ')
+    return text_string.strip()
+
+def getNum(text,start,dist):
+    end=start+1
+    flag=0
+    while start >0 and end-start <=dist and text[start]!='\r' and text[start]!='\n':
+        if text[start].isdigit() and (start-1==0 or text[start-1]==" " or text[start-1]=="\n" or text[start-1]==">" or text[start-1]==")"):
+            flag=1
+            break;
+        start=start-1
+    return flag,start
+
+def getNumNext(text,end,dist):
+    start=end
+    flag=0
+    count=0
+    while end <len(text)-2 and end-start <=dist and text[start]!='\r' and text[start]!='\n':
+        if text[end].isdigit():
+            count+=1
+        if count==5 and text[end+1].isdigit() and (end+1==len(text)-2 or text[end+1]==" " or text[end+1]=="\n" or text[end+1]=="<"):
+            flag=1
+            break;
+        end=end+1
+    return flag,end+1
+
+def getSpace(text,start):
+    while start >0:
+        if text[start-1]==" ":
+            break;
+        start=start-1
+    return start
+
+def extractAddress(text,p,type1,addresses):
+    end=-1
+    m=p.search(text.lower())
+    if m!=None:
+        end=m.span()[0]+len(type1)+1
+    if end !=-1:
+        flag=1
+        flag,bkStart=getNum(text,end-(len(type1)+1),50)
+        if flag==0:
+            start=getSpace(text,end-(len(type1)+2))
+        elif flag==1:
+            flag,start=getNum(text,bkStart-1,10)
+            #print (flag, text[start:end])
+            if flag==0:
+                start=bkStart
+        flag,newEnd=getNumNext(text,end,25)
+        if flag:
+            end=newEnd
+        addresses.add(cleanAddress(text[start:end].lower(),3))
+        m=p.search(text.lower(),end)
+        if m!=None:
+            addresses=extractAddress(text[end:],p,type1,addresses)
+        return addresses
+    return addresses
+#extractAddress("<BR />=?=?=  Table Shower available  =?=?=<br><br>?=Let us make you stress free one day at a time=?<br><br>=?= PUENTE Spa=?= <br><br>TEL:  626-338-8809  <br><br>  1832 Puente Ave. Baldwin Park, CA. 91706  <br><br>Clean Shower Included With Session<br><br>we always hiring beautiful ladies<br><br> Open- 9:00 AM to 9:30 PM<br>  </div>")
+
+
+def processFile(input_file_name,keywords,finalAddressData):
```

Comment:
This code should also be removed since we won't be executing it this way anymore